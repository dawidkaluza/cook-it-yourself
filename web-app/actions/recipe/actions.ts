"use server"

type Recipes = {
  items: [
    id: number,
    name: string,
    description: string
  ]
};

export const getMyRecipes = async () : Promise<Recipes> => {

};
