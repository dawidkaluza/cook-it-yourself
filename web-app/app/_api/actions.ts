"use server"

import {cookies} from "next/headers";

export function isSignedIn(){
  const cookieStore = cookies();
  const sessionCookie = process.env.NEXT_PUBLIC_API_GATEWAY_SESSION_COOKIE;
  return sessionCookie ? cookieStore.has(sessionCookie) : false;
}