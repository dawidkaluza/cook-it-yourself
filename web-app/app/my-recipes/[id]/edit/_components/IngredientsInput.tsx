import React, {useState} from "react";
import {Ingredient} from "@/app/my-recipes/_dtos/recipe";

type IngredientComponentData = Ingredient & {
  key: number;
};

type IngredientsInputProps = {
  errors?: string[];
  ingredients: Ingredient[];
};

const IngredientsInput = (props: IngredientsInputProps) => {
  const initialIngredients = props.ingredients.map((ingredient, index) => {
    return {
      ...ingredient,
      value: Number(ingredient.value).toString(), //temporary workaround
      key: index + 1,
    };
  });

  const [newIngredient, setNewIngredient] = useState("");
  const [ingredients, setIngredients] = useState<IngredientComponentData[]>(initialIngredients);
  const [ingredientsToDelete, setIngredientsToDelete] = useState<number[]>([]);
  const [lastKey, setLastKey] = useState(initialIngredients.length);

  // TODO maybe extract it to a separate hook?
  const generateKey = () => {
    const generatedKey = lastKey + 1;
    setLastKey(generatedKey);
    return generatedKey;
  };

  const addNewIngredient = () => {
    if (newIngredient.trim().length === 0) {
      return;
    }

    const newIngredients = [...ingredients];

    const lastSpaceIndex = newIngredient.lastIndexOf(" ");
    if (lastSpaceIndex === -1 || newIngredient.slice(lastSpaceIndex).search(/ \d/g) !== 0) {
      newIngredients.push({
        name: newIngredient,
        value: "1",
        measure: "",
        key: generateKey(),
      });
    } else {
      const name = newIngredient.slice(0, lastSpaceIndex);
      const fullAmount = newIngredient.slice(lastSpaceIndex + 1);
      const unitIndex = fullAmount.search(/\D/g); // find non-digit character

      if (unitIndex === -1) {
        newIngredients.push({
          name: name,
          value: fullAmount,
          measure: "",
          key: generateKey(),
        });
      } else {
        const amount = fullAmount.slice(0, unitIndex);
        const unit = fullAmount.slice(unitIndex).trim();

        newIngredients.push({
          name: name,
          value: amount,
          measure: unit,
          key: generateKey(),
        });
      }
    }

    setIngredients(newIngredients);
    setNewIngredient("");
  }

  const updateIngredient = (ingredient: IngredientComponentData) => {
    const newIngredients = [ ...ingredients ];
    const index = newIngredients.findIndex(filteredIngredient => filteredIngredient.key === ingredient.key);
    newIngredients[index] = ingredient;
    setIngredients(newIngredients);
  };

  const deleteIngredient = (key: number) => {
    const index = ingredients.findIndex(ingredient => ingredient.key === key);
    const ingredient = ingredients[index];
    if (ingredient.id) {
      const newIngredientsToDelete = [ ...ingredientsToDelete ];
      newIngredientsToDelete.push(ingredient.id);
      setIngredientsToDelete(newIngredientsToDelete);
    }

    setIngredients(
      ingredients.slice(0, index).concat(ingredients.slice(index + 1))
    );
  };

  const errors = props.errors;
  return (
    <div className="row mb-4">
      <label htmlFor="newIngredient" className="col-md-3 col-form-label text-md-end">Ingredients</label>
      <div className="col-md-9">
        <div className="row">
          {ingredientsToDelete.map(ingredientId => (
            <input key={ingredientId.toString()}
                   type="hidden"
                   name="ingredientsToDelete"
                   value={ingredientId.toString()}
            />
          ))}
        </div>

        {ingredients.map(ingredient => (
          <div key={ingredient.key} className="row">
            <IngredientFields
              ingredient={ingredient}
              onUpdate={(ingredient) => updateIngredient(ingredient)}
              onDelete={() => deleteIngredient(ingredient.key)}
            />
          </div>
        ))}
        <div className="row">
          <NewIngredientFields
            value={newIngredient}
            onChange={setNewIngredient}
            onSubmit={addNewIngredient}
          />
        </div>
        <div className="row">
          {errors && errors.map(error => {
            return (
              <div key={error} className="invalid-feedback d-block">
                {error}
              </div>
            )
          })}
        </div>
      </div>
    </div>
  );
};

type IngredientFieldsProps = {
  ingredient: IngredientComponentData;
  onDelete: () => void;
  onUpdate: (ingredient: IngredientComponentData) => void;
};

const IngredientFields = (props: IngredientFieldsProps) => {
  const { ingredient, onDelete, onUpdate } = props;
  const { id } = ingredient;

  return (
    <div className="input-group">
      <input
        type="hidden"
        name="ingredientId"
        value={id}
      />

      <input
        name="ingredientName"
        value={ingredient.name}
        onChange={(event) => onUpdate({ ...ingredient, name: event.target.value })}
        className="form-control"
        style={{minWidth: "50%"}}
        placeholder="Name"
      />

      <input
        name="ingredientValue"
        value={ingredient.value}
        onChange={(event) => onUpdate({ ...ingredient, value: event.target.value })}
        className="form-control"
        style={{minWidth: "15%"}}
        placeholder="Amount"
      />

      <input
        name="ingredientMeasure"
        value={ingredient.measure}
        onChange={(event) => onUpdate({ ...ingredient, measure: event.target.value })}
        className="form-control"
        style={{minWidth: "15%"}}
        placeholder="Unit"
      />

      <button className="btn btn-outline-danger" type="button" onClick={onDelete}>
        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor"
             className="bi bi-dash-lg" viewBox="0 0 16 16">
          <path fillRule="evenodd" d="M2 8a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11A.5.5 0 0 1 2 8"/>
        </svg>
      </button>
    </div>
  );
};

const NewIngredientFields = (
  props: {
    value: string,
    onChange: (value: string) => void,
    onSubmit: () => void,
  }
) => {
  const { value, onChange, onSubmit } = props;
  return (
    <div className="input-group">
      <input
        name="newIngredient" id="newIngredient"
        className="form-control"
        value={value}
        placeholder="e.g., water, apple 2, sugar 250g"
        onChange={(e) => onChange(e.currentTarget.value)}
        onKeyDown={(e) => {
          if (e.key === 'Enter') {
            e.preventDefault();
            onSubmit();
          }
        }}
      />
      <button className="btn btn-success" type="button" onClick={onSubmit}>
        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor"
             className="bi bi-plus-lg" viewBox="0 0 16 16">
          <path fillRule="evenodd"
                d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2"/>
        </svg>
      </button>
    </div>
  );
};

export {IngredientsInput};