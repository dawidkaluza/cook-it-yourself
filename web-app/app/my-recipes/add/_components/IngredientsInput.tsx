"use client";

import React, {useState} from "react";
import {FieldError} from "@/app/my-recipes/add/actions";

type Props = {
  fieldErrors?: FieldError[];
};

type Ingredient = {
  id: number;
  name: string;
  amount: number;
  unit: string;
};

const IngredientsInput = ({ fieldErrors }: Props) => {
  const [newIngredient, setNewIngredient] = useState("");
  const [ingredients, setIngredients] = useState<Ingredient[]>([]);
  const [generatedId, setGeneratedId] = useState(0);

  const generateNewId = () => {
    const newId = generatedId + 1;
    setGeneratedId(newId);
    return newId;
  }

  const addNewIngredient = () => {
    if (newIngredient.trim().length === 0) {
      return;
    }

    const newIngredients = [...ingredients];
    const newIngredientId = generateNewId();

    const lastSpaceIndex = newIngredient.lastIndexOf(" ");
    if (lastSpaceIndex === -1 || newIngredient.slice(lastSpaceIndex).search(/ \d/g) !== 0) {
      newIngredients.push({
        id: newIngredientId,
        name: newIngredient,
        amount: 1,
        unit: "",
      });
    } else {
      const name = newIngredient.slice(0, lastSpaceIndex);
      const fullAmount = newIngredient.slice(lastSpaceIndex + 1);
      const unitIndex = fullAmount.search(/\D/g); // find non-digit character

      if (unitIndex === -1) {
        newIngredients.push({
          id: newIngredientId,
          name: name,
          amount: Number(fullAmount),
          unit: "",
        });
      } else {
        const amount = Number(fullAmount.slice(0, unitIndex));
        const unit = fullAmount.slice(unitIndex);

        newIngredients.push({
          id: newIngredientId,
          name: name,
          amount: amount,
          unit: unit,
        });
      }
    }

    setIngredients(newIngredients);
    setNewIngredient("");
  }

  const deleteIngredient = (id: number) => {
    setIngredients(ingredients.filter((ingredient) => ingredient.id !== id));
  };

  return (
    <div className="row mb-4">
      <label htmlFor="newIngredient" className="col-sm-2 col-form-label">Ingredients</label>
      <div className="col-sm-10">
        {ingredients.map(ingredient => (
          <div key={ingredient.id} className="row">
            <IngredientFields
              ingredient={ingredient}
              onDelete={() => deleteIngredient(ingredient.id)}
            />
          </div>
        ))}
        <div className="row">
          <NewIngredientFields
            value={newIngredient}
            onChange={setNewIngredient}
            onSubmit={addNewIngredient}
          />
        </div>
        <div className="row">
          {fieldErrors && fieldErrors.map(fieldError => {
            return (
              <div key={fieldError.message} className="invalid-feedback d-block">
                {fieldError.message}
              </div>
            )
          })}
        </div>
      </div>
    </div>
  );
};

const IngredientFields = (props: { ingredient: Ingredient, onDelete: () => void }) => {
  const { ingredient, onDelete } = props;
  const [ name, setName ] = useState(ingredient.name);
  const [ amount, setAmount ] = useState(ingredient.amount.toString());
  const [ unit, setUnit ] = useState(ingredient.unit);

  return (
    <div className="input-group">
      <input
        name="ingredientName"
        value={name}
        onChange={(event) => setName(event.target.value)}
        className="form-control"
        placeholder="Name" style={{minWidth: "60%"}}
      />

      <input
        name="ingredientAmount"
        value={amount}
        onChange={(event) => setAmount(event.target.value)}
        className="form-control"
        placeholder="Amount" style={{minWidth: "20%"}}
      />

      <input
        name="ingredientUnit"
        value={unit}
        onChange={(event) => setUnit(event.target.value)}
        className="form-control"
        placeholder="Unit"
      />

      <button className="btn btn-outline-danger" type="button" onClick={onDelete}>
        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor"
           className="bi bi-dash-lg" viewBox="0 0 16 16">
          <path fillRule="evenodd" d="M2 8a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11A.5.5 0 0 1 2 8"/>
        </svg>
      </button>
    </div>
  );
};

const NewIngredientFields = (
  props: {
    value: string,
    onChange: (value: string) => void,
    onSubmit: () => void,
  }
) => {
  const { value, onChange, onSubmit } = props;
  return (
    <div className="input-group">
      <input
        name="newIngredient" id="newIngredient" className="form-control"
        value={value}
        placeholder="Name and optionally amount (paprikas 3, sugar 250g, etc.)"
        onChange={(e) => onChange(e.currentTarget.value)}
        onKeyDown={(e) => {
          if (e.key === 'Enter') {
            e.preventDefault();
            onSubmit();
          }
        }}
      />
      <button className="btn btn-success" type="button" onClick={onSubmit}>
        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor"
             className="bi bi-plus-lg" viewBox="0 0 16 16">
          <path fillRule="evenodd"
                d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2"/>
        </svg>
      </button>
    </div>
  );
};

export {IngredientsInput};