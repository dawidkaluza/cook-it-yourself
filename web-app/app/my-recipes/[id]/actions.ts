"use server";

import {RecipeDetails} from "@/app/my-recipes/_dtos/recipe";
import { fetchFromServer} from "@/app/_api/fetch";

export async function reviewRecipe(id: number) : Promise<RecipeDetails> {
  return await fetchFromServer<RecipeDetails>({
    endpoint: `/kitchen/recipe/${id}`
  }) as RecipeDetails;
}