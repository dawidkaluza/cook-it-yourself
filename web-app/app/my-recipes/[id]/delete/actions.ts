"use server";

import {ApiError, fetchFromServer} from "@/app/_api/fetch";
import {redirect} from "next/navigation";

export async function deleteRecipe(prevState: any, formData: FormData): Promise<string> {
  const id = formData.get("id");

  try {
    await fetchFromServer({
      endpoint: `/kitchen/recipe/${id}`,
      method: "DELETE"
    });
  } catch (error) {
    if (error instanceof ApiError) {
      const response = error.response;
      switch (response.status) {
        case 404: {
          return "The recipe is already deleted.";
        }

        case 422: {
          return "Couldn't process the request.";
        }
      }
    }

    throw error;
  }

  redirect("/my-recipes");
}