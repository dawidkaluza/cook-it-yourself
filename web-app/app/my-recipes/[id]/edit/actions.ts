"use server";

import {FieldError} from "@/app/my-recipes/_dtos/errors";
import {ApiError, fetchFromServer} from "@/app/_api/fetch";
import {PortionSize, RecipeDetails, UpdateRecipeRequest} from "@/app/my-recipes/_dtos/recipe";
import {redirect} from "next/navigation";

async function editRecipe(prevState: any, formData: FormData): Promise<FieldError[]> {
  const id = formData.get("id") as string;

  const mapToPortionSize = (formData: FormData): PortionSize => {
    const portionSizeInput = formData?.get("portionSize")?.toString();
    if (!portionSizeInput) {
      return {
        value: "",
        measure: "",
      };
    }

    const numberIndex = portionSizeInput.search(/\d/g);
    if (numberIndex === -1) {
      return {
        value: "",
        measure: "",
      };
    }

    const nonNumberIndex = portionSizeInput.search(/\D/g);
    if (nonNumberIndex === -1 || nonNumberIndex < numberIndex) {
      return {
        value: portionSizeInput.slice(numberIndex),
        measure: "",
      };
    }

    return {
      value: portionSizeInput.slice(numberIndex, nonNumberIndex).trim(),
      measure: portionSizeInput.slice(nonNumberIndex).trim()
    };
  }

  const requestData: UpdateRecipeRequest = {
    basicInformation: {
      name: formData.get("name") as string,
      description: formData.get("description") as string,
      cookingTime: Number(formData.get("cookingTime")),
      portionSize: mapToPortionSize(formData),
    },
    ingredients: {
      ingredientsToAdd: [],
      ingredientsToUpdate: [],
      ingredientsToDelete: [],
    },
    steps: {
      stepsToAdd: [],
      stepsToUpdate: [],
      stepsToDelete: [],
    }
  };

  try {
    const updatedRecipe = await fetchFromServer<RecipeDetails>({
      endpoint: `/kitchen/recipe/${id}`,
      method: "PUT",
      body: requestData
    });

    if (updatedRecipe) {
      redirect(`/my-recipes/${updatedRecipe.id}/`);
    }
  } catch (error) {
    if (error instanceof ApiError) {
      const response = error.response;

      // TODO cover other possible client errors
      if (response.status === 422) {
        const body = await response.json();
        const fields : FieldError[] = body.fields;
        return fields.map((field) => {
          const name = field.name;
          const message = field.message;
          const indexOfDot = name.indexOf(".");
          if (indexOfDot === -1) {
            return {
              name,
              message
            }
          } else {
            const mainName = name.slice(0, indexOfDot);
            return {
              name: mainName,
              message
            };
          }
        });
      }

      return [];
    }

    throw error;
  }

  throw new Error("Couldn't process the request");
}

export { editRecipe };