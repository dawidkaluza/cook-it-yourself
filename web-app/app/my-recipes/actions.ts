"use server";

import {fetchFromServer} from "@/app/_api/fetch";
import {Page} from "@/app/my-recipes/_dtos/page";
import {Recipe} from "@/app/my-recipes/_dtos/recipe";

export async function getMyRecipes(){
  return await fetchFromServer({ endpoint: "/kitchen/recipe" }) as Promise<Page<Recipe>>;
}