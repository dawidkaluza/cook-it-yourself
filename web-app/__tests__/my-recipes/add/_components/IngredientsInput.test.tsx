import {describe, expect, test} from "vitest";
import {IngredientsInput} from "@/app/my-recipes/add/_components/IngredientsInput";
import {fireEvent, render, screen, within} from "@testing-library/react";
import {FieldError} from "@/app/my-recipes/add/actions";

describe("IngredientsInput component", () => {
  test("render without errors", () => {
    // Given
    // When
    const component = render(<IngredientsInput />);

    // Then
    const inputElement = screen.getByLabelText("Ingredients");
    expect(inputElement).not.toBeNull();

    expect(component.container).toMatchSnapshot();
    component.unmount();
  });

  test("render with errors", () => {
    // Given
    const fieldErrors: FieldError[] = [
      {
        name: "ingredient",
        message: "Some ingredient error.",
      },
      {
        name: "ingredient",
        message: "Some other ingredient error.",
      },
    ];

    // When
    const component = render(<IngredientsInput fieldErrors={fieldErrors} />);

    // Then
    const inputElement = screen.getByLabelText("Ingredients");
    expect(inputElement).not.toBeNull();

    for (const fieldError of fieldErrors) {
      const errorElement = screen.getByText(fieldError.message);
      expect(errorElement).not.toBeNull();
    }

    expect(component.container).toMatchSnapshot();
    component.unmount();
  });

  test("render and add new ingredient", () => {
    // Given
    const expectedInputs = [
      "Sausages",
      "1",
      "Pot",
      "2",
      "Water",
      "500",
      "ml"
    ];

    const component = render(<IngredientsInput />);

    // When
    const newIngredientInputEl = screen.getByLabelText("Ingredients");
    const addButtonEl = screen.getByRole("button");

    fireEvent.change(newIngredientInputEl, { target: { value: "" } });
    fireEvent.click(addButtonEl);

    fireEvent.change(newIngredientInputEl, { target: { value: "Sausages" } });
    fireEvent.click(addButtonEl);

    fireEvent.change(newIngredientInputEl, { target: { value: "Pot 2" } });
    fireEvent.click(addButtonEl);

    fireEvent.change(newIngredientInputEl, { target: { value: "Water 500ml" } });
    fireEvent.keyDown(newIngredientInputEl, {key: 'Enter', code: 'Enter', charCode: 13});

    // Then
    const allInputElements = screen.getAllByRole("textbox");

    for (const expectedInput of expectedInputs) {
      expect(allInputElements.filter(
        el => el.getAttribute("value") === expectedInput
      )).toHaveLength(1);
    }

    expect(component.container).toMatchSnapshot();
    component.unmount();
  });

  test("render, add and modify added ingredients", () => {
    // Given
    const expectedInputs = [
      "Sausage",
      "1",
      "Pot",
      "3",
      "Water",
      "500",
      "g"
    ];

    const component = render(<IngredientsInput />);

    const newIngredientInputEl = screen.getByLabelText("Ingredients");
    const addButtonEl = screen.getByRole("button");

    fireEvent.change(newIngredientInputEl, { target: { value: "Sausages" } });
    fireEvent.click(addButtonEl);

    fireEvent.change(newIngredientInputEl, { target: { value: "Pot 2" } });
    fireEvent.click(addButtonEl);

    fireEvent.change(newIngredientInputEl, { target: { value: "Water 500ml" } });
    fireEvent.click(addButtonEl);

    // When
    const sausagesIngredientEl = screen.getByDisplayValue("Sausages");
    fireEvent.change(sausagesIngredientEl, { target: { value: "Sausage" } });

    const potAmountEl = screen.getByDisplayValue("2");
    fireEvent.change(potAmountEl, { target: { value: "3" } });

    const waterUnitEl = screen.getByDisplayValue("ml");
    fireEvent.change(waterUnitEl, { target: { value: "g" } });

    fireEvent.click(addButtonEl); // workaround to cause state update

    // Then
    const allInputElements = screen.getAllByRole("textbox");

    for (const expectedInput of expectedInputs) {
      console.log("Input: ", expectedInput);
      expect(allInputElements.filter(
        el => el.getAttribute("value") === expectedInput
      )).toHaveLength(1);
    }

    expect(component.container).toMatchSnapshot();
    component.unmount();
  });

  test("render, add and remove ingredients", () => {
    // Given
    const expectedInputs = [
      "Pot",
      "2",
      "Water",
      "500",
      "ml"
    ];

    const expectedRemovedInputs = [
      "Sausages",
    ];

    const component = render(<IngredientsInput />);

    const newIngredientInputEl = screen.getByLabelText("Ingredients");
    const addButtonEl = screen.getByRole("button");

    fireEvent.change(newIngredientInputEl, { target: { value: "Sausages" } });
    fireEvent.click(addButtonEl);

    fireEvent.change(newIngredientInputEl, { target: { value: "Pot 2" } });
    fireEvent.click(addButtonEl);

    fireEvent.change(newIngredientInputEl, { target: { value: "Water 500ml" } });
    fireEvent.click(addButtonEl);

    // When
    const sausagesDeleteBtnEl = within(
      screen.getByDisplayValue("Sausages").parentElement as HTMLElement
    ).getByRole("button");
    fireEvent.click(sausagesDeleteBtnEl);

    // Then
    const allInputElements = screen.getAllByRole("textbox");

    for (const expectedInput of expectedInputs) {
      expect(allInputElements.filter(
        el => el.getAttribute("value") === expectedInput
      )).toHaveLength(1);
    }

    for (const expectedRemovedInput of expectedRemovedInputs) {
      expect(allInputElements.filter(
        el => el.getAttribute("value") === expectedRemovedInput
      )).toHaveLength(0);
    }

    expect(component.container).toMatchSnapshot();
    component.unmount();
  });
});