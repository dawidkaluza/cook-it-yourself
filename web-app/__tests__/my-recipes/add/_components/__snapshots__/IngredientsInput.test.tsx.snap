// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IngredientsInput component > render and add new ingredient 1`] = `
<div>
  <div
    class="row mb-4"
  >
    <label
      class="col-sm-2 col-form-label"
      for="newIngredient"
    >
      Ingredients
    </label>
    <div
      class="col-sm-10"
    >
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            name="ingredientName"
            placeholder="Name"
            style="min-width: 60%;"
            value="Sausages"
          />
          <input
            class="form-control"
            name="ingredientAmount"
            placeholder="Amount"
            style="min-width: 20%;"
            value="1"
          />
          <input
            class="form-control"
            name="ingredientUnit"
            placeholder="Unit"
            value=""
          />
          <button
            class="btn btn-outline-danger"
            type="button"
          >
            <svg
              class="bi bi-dash-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 8a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11A.5.5 0 0 1 2 8"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            name="ingredientName"
            placeholder="Name"
            style="min-width: 60%;"
            value="Pot"
          />
          <input
            class="form-control"
            name="ingredientAmount"
            placeholder="Amount"
            style="min-width: 20%;"
            value="2"
          />
          <input
            class="form-control"
            name="ingredientUnit"
            placeholder="Unit"
            value=""
          />
          <button
            class="btn btn-outline-danger"
            type="button"
          >
            <svg
              class="bi bi-dash-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 8a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11A.5.5 0 0 1 2 8"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            name="ingredientName"
            placeholder="Name"
            style="min-width: 60%;"
            value="Water"
          />
          <input
            class="form-control"
            name="ingredientAmount"
            placeholder="Amount"
            style="min-width: 20%;"
            value="500"
          />
          <input
            class="form-control"
            name="ingredientUnit"
            placeholder="Unit"
            value="ml"
          />
          <button
            class="btn btn-outline-danger"
            type="button"
          >
            <svg
              class="bi bi-dash-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 8a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11A.5.5 0 0 1 2 8"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="newIngredient"
            name="newIngredient"
            placeholder="Name and optionally amount (paprikas 3, sugar 250g, etc.)"
            value=""
          />
          <button
            class="btn btn-success"
            type="button"
          >
            <svg
              class="bi bi-plus-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      />
    </div>
  </div>
</div>
`;

exports[`IngredientsInput component > render with errors 1`] = `
<div>
  <div
    class="row mb-4"
  >
    <label
      class="col-sm-2 col-form-label"
      for="newIngredient"
    >
      Ingredients
    </label>
    <div
      class="col-sm-10"
    >
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="newIngredient"
            name="newIngredient"
            placeholder="Name and optionally amount (paprikas 3, sugar 250g, etc.)"
            value=""
          />
          <button
            class="btn btn-success"
            type="button"
          >
            <svg
              class="bi bi-plus-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="invalid-feedback d-block"
        >
          Some ingredient error.
        </div>
        <div
          class="invalid-feedback d-block"
        >
          Some other ingredient error.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`IngredientsInput component > render without errors 1`] = `
<div>
  <div
    class="row mb-4"
  >
    <label
      class="col-sm-2 col-form-label"
      for="newIngredient"
    >
      Ingredients
    </label>
    <div
      class="col-sm-10"
    >
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="newIngredient"
            name="newIngredient"
            placeholder="Name and optionally amount (paprikas 3, sugar 250g, etc.)"
            value=""
          />
          <button
            class="btn btn-success"
            type="button"
          >
            <svg
              class="bi bi-plus-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      />
    </div>
  </div>
</div>
`;

exports[`IngredientsInput component > render, add and remove ingredients 1`] = `
<div>
  <div
    class="row mb-4"
  >
    <label
      class="col-sm-2 col-form-label"
      for="newIngredient"
    >
      Ingredients
    </label>
    <div
      class="col-sm-10"
    >
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            name="ingredientName"
            placeholder="Name"
            style="min-width: 60%;"
            value="Pot"
          />
          <input
            class="form-control"
            name="ingredientAmount"
            placeholder="Amount"
            style="min-width: 20%;"
            value="2"
          />
          <input
            class="form-control"
            name="ingredientUnit"
            placeholder="Unit"
            value=""
          />
          <button
            class="btn btn-outline-danger"
            type="button"
          >
            <svg
              class="bi bi-dash-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 8a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11A.5.5 0 0 1 2 8"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            name="ingredientName"
            placeholder="Name"
            style="min-width: 60%;"
            value="Water"
          />
          <input
            class="form-control"
            name="ingredientAmount"
            placeholder="Amount"
            style="min-width: 20%;"
            value="500"
          />
          <input
            class="form-control"
            name="ingredientUnit"
            placeholder="Unit"
            value="ml"
          />
          <button
            class="btn btn-outline-danger"
            type="button"
          >
            <svg
              class="bi bi-dash-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 8a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11A.5.5 0 0 1 2 8"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="newIngredient"
            name="newIngredient"
            placeholder="Name and optionally amount (paprikas 3, sugar 250g, etc.)"
            value=""
          />
          <button
            class="btn btn-success"
            type="button"
          >
            <svg
              class="bi bi-plus-lg"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row"
      />
    </div>
  </div>
</div>
`;
